lib/console/organization_hotspots/organization_hotspots.ex

```diff
@@ -34,4 +34,28 @@ defmodule Console.OrganizationHotspots do
   def delete_org_hotspot(%OrganizationHotspot{} = org_hotspot) do
     Repo.delete(org_hotspot)
   end
+
+  def claim_org_hotspots(hotspot_addresses, organization) do
+    new_organization_hotspots = Enum.reduce(hotspot_addresses, [], fn hotspot_address, acc ->
+      if Repo.get_by(OrganizationHotspot, hotspot_address: hotspot_address, organization_id: organization.id) == nil do
+        acc ++ [%{ hotspot_address: hotspot_address, organization_id: organization.id, claimed: true }]
+      else
+        acc
+      end
+    end)
+
+    with {:ok, :ok} <- Repo.transaction(fn ->
+        Enum.each(new_organization_hotspots, fn attrs ->
+          Repo.insert!(OrganizationHotspot.changeset(%OrganizationHotspot{}, attrs))
+        end)
+      end)
+    do
+      {:ok, length(new_organization_hotspots), new_organization_hotspots}
+    end
+  end
+
+  def unclaim_org_hotspots(hotspot_addresses, organization) do
+    query = from(oh in OrganizationHotspot, where: oh.hotspot_address in ^hotspot_addresses and oh.organization_id == ^organization.id)
```

Comment:
we dont want to simply delete them all since we could have a row that isn't claimed but has the alias set

Reply:
ah! good catch. will update.